pytests/helpers/webhelper.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root
+# for license information.
+
+from contextlib import contextmanager
+import threading
+import requests
+import re
+
+def get_web_string(path, obj):
+    r = requests.get(path)
+    content = r.text
+    if obj is not None:
+        obj['content'] = content
+    return content
+
+
+def get_web_string_no_error(path, obj):
+    try:
+        return get_web_string(path, obj)
+    except Exception:
+        pass
+
+
+re_link = r"(http(s|)\:\/\/[\w\.]*\:[0-9]{4,6}(\/|))"
+def get_url_from_str(s):
+    matches = re.findall(re_link, s)
+    if matches and matches[0]and matches[0][0].strip():
+        return matches[0][0]
+    return None
+
+
+@contextmanager
+def get_web_content(link, web_result=None, timeout=1):
+    web_client_thread = threading.Thread(
+        target=get_web_string_no_error,
+        args=(link, web_result),
+        name='test.webClient'
+    )
+    web_client_thread.start()
+    yield web_result
```

Comment:
I think this helper looks somewhat confusing as a context manager - on the first glance, it feels like `web_result` is usable inside the with-statement, but in practice you can only get 'content' after the background thread has ended its execution, i.e. after with-statement. 

What this actually reminds me most of is `send_request` / `wait_for_response` - the part before `yield` is send, and the part after is `wait`. Maybe make it an object that works like that? i.e. something like:

```py
def get_web_content(link, web_result=None, timeout=1):
  class WebResponse(object):
    def __init__(self):
       self.content = None

    def wait_for_response(self):
       web_client_thread.join()
       return self.content

  response = WebResponse()
  web_client_thread = threading.Thread(
        target=get_web_string_no_error,
        args=(link, response),
        name='test.webClient'
  )
  return response
```
and change `get_web_string` so that it does `obj.content = ...` instead of `obj['content'] = ...'.

Then the test would do:
```py
web_request = get_web_content(...)
# Wait for breakpoint being hit etc. 
...
content = web_request.wait_for_response()
# Do asserts based on content
```

Reply:
That is better will change.